zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/impl/log/LogElement.kt

```diff
@@ -0,0 +1,128 @@
+package org.hexworks.zircon.internal.component.impl.log
+
+import org.hexworks.zircon.api.builder.data.TileBuilder
+import org.hexworks.zircon.api.component.Component
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.data.Size
+import org.hexworks.zircon.api.graphics.SubTileGraphics
+import org.hexworks.zircon.api.modifier.Delay
+import org.hexworks.zircon.api.modifier.Modifier
+import org.hexworks.zircon.internal.component.renderer.LogElementRenderContext
+import org.hexworks.zircon.internal.component.renderer.LogElementRenderResult
+import org.hexworks.zircon.internal.component.renderer.VisibleRenderArea
+
+sealed class LogElement(private val xPosition: Int) {
+    var modifiers: Set<Modifier> = setOf()
+    var screenPositionArea: ScreenPositionArea? = null
+    lateinit var logElementRow: LogElementRow
+
+    abstract fun getSize(): Size
+    abstract fun render(tileGraphics: SubTileGraphics, context: LogElementRenderContext): LogElementRenderResult
+
+    fun getPosition() = Position.create(xPosition, logElementRow.yPosition)
+
+}
+
+data class LogTextElement(val text: String, private val xPosition: Int) : LogElement(xPosition) {
```

Comment:
I think this class is redundant: there is [CharacterTileString](https://github.com/Hexworks/zircon/blob/master/zircon.core/src/main/kotlin/org/hexworks/zircon/api/graphics/CharacterTileString.kt) which was written for this purpose. You can also use a `TextBox` (the builder is [here](https://github.com/Hexworks/zircon/blob/master/zircon.core/src/main/kotlin/org/hexworks/zircon/api/builder/component/TextBoxBuilder.kt)) which supports setting `contentWidth` in the builder and it will automatically wrap the text for you.